src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String result = getStringFromFile(fromFileName);
+        Path outputFile = Paths.get(toFileName);
+        try {
+            Files.write(outputFile, result.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException ioException) {
+            throw new RuntimeException(ioException);
+        }
+    }
+
+    private String getStringFromFile(String fromFileName) {
+        Path inputFile = Paths.get(fromFileName);
+        final int[] supply = {0};
+        final int[] buy = {0};
+        final int indexOperation = 0;
+        final int indexValue = 1;
 
+        try (Stream<String> lines = Files.lines(inputFile)) {
+            lines.forEach(line -> {
+                String[] parts = line.split(",");
+                if ("supply".equals(parts[indexOperation])) {
+                    supply[0] += Integer.parseInt(parts[indexValue]);
+                } else if ("buy".equals(parts[indexOperation])) {
+                    buy[0] += Integer.parseInt(parts[indexValue]);
+                }
+            });
+        } catch (IOException ioException) {
+            throw new RuntimeException(ioException);
+        }
+        String result = String.format("supply,%d%nbuy,%d%nresult,%d", supply[0], buy[0],
+                supply[0] - buy[0]);
+        return result;
```

Comment:
Instead of using arrays to hold the values of `supply` and `buy`, you should use local variables since they don't need to be final or effectively final. Arrays are used here to bypass the final or effectively final requirement for lambda, which is a sign that the lambda should likely be refactored into a method.

Reply:
Local variables can not change at lambda!